### PR TITLE
docs: Add example of contract-out

### PR DESCRIPTION
**Pull request by ZachOBrien**
Suggestion from the Racketcon hackathon

### CODE REVIEW

File: pkgs/racket-doc/scribblings/reference/contracts.scrbl [hunk 1]
```diff
@@ -1944,6 +1944,22 @@ is bound to vectors of two elements, the exported identifier and a
...
+
+ (require racket/contract)
+
+ (provide
+  (contract-out
+   [recip (-> (and/c real? non-zero?)
+              (between/c -1 1))]))
+
+ (define (recip x) (/ 1 x))
+ (define (non-zero? x) (not (= x 0)))
```
Comment (jessealama) [inline pkgs/racket-doc/scribblings/reference/contracts.scrbl:1960]: Perhaps `(zero? x)`?

Comment (jessealama) [reply pkgs/racket-doc/scribblings/reference/contracts.scrbl:1960]: Or perhaps, using `not/c`:

````
(and/c real? (not/c zero?))
```

And then you can drop `(define (non-zero? x) ...)`.

Comment (ZachOBrien) [reply pkgs/racket-doc/scribblings/reference/contracts.scrbl:1960]: Thanks, good suggestion

**Comment by sorawee** [comment]
I'd suggest using `@examples`. See `recontract-out` for an example.

**Comment by ZachOBrien** [comment]
Thanks, I'll look into examples.

I took this contract from a nearby example in the documentation for consistency's sake, but it turns out it's incorrect... Given a real number less than 1, the reciprocal will be outside range [-1, 1]. So there is a bug in the existing documentation here:

https://github.com/racket/racket/blob/0b8732d9666044be6786dcbf5ca651c41eaed7d8/pkgs/racket-doc/scribblings/reference/contracts.scrbl#L1965-L1971

**Pull request merged**